src/modules/products/products.controller.ts

```diff
@@ -0,0 +1,58 @@
+import { Controller, Post, Body, Req, UseGuards, HttpException, InternalServerErrorException } from '@nestjs/common';
+import { ProductsService } from './products.service';
+import { CreateProductDto } from './dto/create-product.dto';
+import {
+  ApiTags,
+  ApiBody,
+  ApiOperation,
+  ApiOkResponse,
+  ApiInternalServerErrorResponse,
+  ApiNotFoundResponse,
+} from '@nestjs/swagger';
+import { skipAuth } from '../../helpers/skipAuth';
+
+@Controller('products')
+@ApiTags('Products')
+export class ProductController {
+  constructor(private readonly productsService: ProductsService) {}
+
+  @Post()
+  @skipAuth()
```

Comment:
Should this be a public route, coz adding the skipAuth decoratior bypasses authentication.
Is this route public or protected. If protected, remove the skipAuth decorator